src/tooling/debugging/vscode-debugging.md

```diff
@@ -0,0 +1,136 @@
+
+# Debugging in Visual Studio Code
+
+There is also a possibility to debug with graphical output directly in Visual Studio Code.
+
+# ESP32
+
+## Hardware Setup
+
+ESP32 doesn't have a built-in JTAG interface so you have to connect an external JTAG adapter with the ESP32 board, for example [ESP-Prog](https://docs.espressif.com/projects/espressif-esp-iot-solution/en/latest/hw-reference/ESP-Prog_guide.html) can be used.
+
+|   ESP32 Pin   |   JTAG Signal   |
+| :-----------: | :-------------: |
+| MTDO/GPIO15   |      TDO        |
+| MTDI/GPIO12   |      TDI        |
+| MTCK/GPIO13   |      TCK        |
+| MTMS/GPIO14   |      TMS        |
+| 3V3           |      VJTAG      |
+| GND           |      GND        |
+
+
+**Note**: On Windows `USB Serial Converter A 0403 6010 00` driver should be WinUSB.
+
+## Set up VSCode
+
+1. Install [Cortex-Debug](https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug) extension for VScode.
+2. Create the `.vscode/launch.json` file in the project tree you want to debug. [This](https://github.com/esp-rs/esp32-hal/blob/master/.vscode/launch.json) can be used as a template file.
+3. Update **executable**, **svdFile** paths and **toolchainPrefix** field.
```

Comment:
They also need to change serverpath to the xtensa enabled espressif one

Reply:
That's good catch, thanks @MabezDev !